packages/viz/__tests__/styles/ColorBinsStyle.spec.ts

```diff
@@ -49,15 +49,73 @@ describe('ColorBinsStyle', () => {
         breaks: [20, 50],
         bins: 1
       });
-      expect(() => style.getLayerProps(styledLayer)).toThrow(CartoStylingError);
+
+      try {
+        style.getLayerProps(styledLayer);
+      } catch (error) {
+        expect(error).toBeInstanceOf(CartoStylingError);
+      }
     });
 
     it('should launch styling error when breaks and palette missmatch', () => {
       const style = colorBinsStyle(FIELD_NAME, {
         breaks: [20, 50],
         palette: ['#ff0', '#231']
       });
-      expect(() => style.getLayerProps(styledLayer)).toThrow(CartoStylingError);
+
+      try {
+        style.getLayerProps(styledLayer);
+      } catch (error) {
+        expect(error).toBeInstanceOf(CartoStylingError);
+      }
+    });
+
+    it('should fails with invalid bins', () => {
+      const style = colorBinsStyle(FIELD_NAME, {
+        bins: 0
```

Comment:
This is controlled nicely. IMO we should now do the same for the **upper limit**.

I have just tried with `style-helpers/color-bins-style.html` and bins: 10 and it fails, because we're running out of colors.

Having a look at cartocolors ramps length (quantitative ramps, valid for numerical fields), a proper limit can be set on 7 (included). In Builder, valid number of buckets is indeed 2 to 7.

So let's try to control that too

Reply:
oki